pkg/reconciler/taskrun/validate_taskrun.go

```diff
@@ -30,58 +30,41 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params []v1beta1.Param, matrix *v1beta1.Matrix) error {
+func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params v1beta1.Params, matrixAllParams v1beta1.Params) error {
```

Comment:
nit: `matrixAllParams` is a bit redundant/awkward, I'd call this `matrixParams` and add a docstring